var/spack/repos/builtin/packages/xz/package.py

```diff
@@ -63,7 +70,60 @@ def determine_version(cls, exe):
         match = re.search(r"xz \(XZ Utils\) (\S+)", output)
         return match.group(1) if match else None
 
+
+class AutotoolsBuilder(AutotoolsBuilder):
+    def configure_args(self):
+        return self.enable_or_disable("libs")
+
     @run_after("install")
     def darwin_fix(self):
         if self.spec.satisfies("platform=darwin"):
             fix_darwin_install_name(self.prefix.lib)
+
+
+class MSBuildBuilder(MSBuildBuilder):
+    @property
+    def build_directory(self):
+        def get_file_string_number(f):
+            s = re.findall(r"\d+$", f)
+            return (int(s[0]) if s else -1, f)
+
+        win_dir = os.path.join(super().build_directory, "windows")
+        compiler_dirs = []
+        with working_dir(win_dir):
+            for obj in os.scandir():
+                if obj.is_dir():
+                    compiler_dirs.append(obj.name)
+        newest_compiler = max(compiler_dirs, key=get_file_string_number)
+        return os.path.join(win_dir, newest_compiler)
+
+    def is_64bit(self):
+        return "64" in self.pkg.spec.target.family
+
+    def msbuild_args(self):
+        plat = "x64" if self.is_64bit() else "x86"
+        if self.pkg.spec.satisfies("libs=shared,static"):
+            f = "xz_win.sln"
+        elif self.pkg.spec.satisfies("libs=shared"):
+            f = "liblzma_dll.vcxproj"
+        else:
+            f = "liblzma.vcxproj"
+        return [self.define("Configuration", "Release"), self.define("Platform", plat), f]
+
+    def install(self, pkg, spec, prefix):
+        with working_dir(self.build_directory):
+            # Ensure we have libs directory
+            mkdirp(prefix.lib)
+            libs_to_find = []
+            if "shared" in self.pkg.spec:
```

Comment:
I think this has to be `if libs=shared in self.pkg.spec`

Reply:
Would `shared` appear elsewhere in the spec? If so you're correct, otherwise I think its fine.

Reply:
This isn't an issue of potential ambiguity, this check will always be false even for `xz libs=shared`, because variant queries are either of the form:

`if "+varname" in spec`

or

`if "varname=..." in spec`

other forms are not valid variant queries (you can try this in `spack-python`). We should probably be raising an error with this query, but for some ambiguous queries it would be impossible to determine whether the query was truly an error. In particular, if you have a dag like `x -> y` you can ask `if 'y' in specx`.

Reply:
Oh I see, I always assume those checks were performed by simple string parsing, I never realized it was actually a standardized query, thanks for taking the time to explain that!

I've been formatting them correctly up to this point purely by assuming it was a style convention (which would have necessitated this change anyway)